test/services/homework/index.test.js

```diff
@@ -402,115 +437,148 @@ describe('homework service', () => {
-						submitted: true,
-						graded: true,
-					})
-				);
+				const schoolId = result.data[0].schoolId.toString();
+				const courseSchoolId = result.data[0].courseId.schoolId.toString();
+				expect(schoolId).to.equal(courseSchoolId);
 			});
-			await Promise.all(submissionPromises);
-		};
-		it('does not paginate the submissions', async () => {
-			const { _id: teacherId } = await testObjects.createTestUser({ roles: ['teacher'] });
-			const student = await testObjects.createTestUser({ roles: ['student'] });
-			const otherStudentPromises = [];
-			for (let i = 0; i < 25; i += 1) {
-				otherStudentPromises.push(testObjects.createTestUser({ roles: ['student'] }));
-			}
-			let otherStudents = await Promise.all(otherStudentPromises);
-			otherStudents = otherStudents.map((s) => s._id);
-			const { _id: courseId } = await testObjects.createTestCourse({
-				teacherIds: [teacherId],
-				userIds: [student._id, ...otherStudents],
-			});
-
-			const homeworkPromises = [];
-			for (let i = 0; i < 50; i += 1) {
-				homeworkPromises.push(createHomeworkWithSubmissions(teacherId, [student._id, ...otherStudents], courseId));
-			}
-			await Promise.all(homeworkPromises);
 
-			const params = await testObjects.generateRequestParamsFromUser(student);
-			const result = await homeworkService.find(params);
-
-			expect(result.data.length).to.equal(50);
-			result.data.forEach((homework) => {
-				expect(homework.submissions).to.equal(1);
+			const createHomeworkWithSubmissions = async (teacherId, studentIds = [], courseId) => {
+				const { _id: homeworkId } = await testObjects.createTestHomework({
+					teacherId,
+					name: 'Testaufgabe',
+					description: '\u003cp\u003eAufgabenbeschreibung\u003c/p\u003e\r\n',
+					availableDate: Date.now(),
+					dueDate: Date.now() + 86400000,
+					lessonId: null,
+					courseId,
+				});
+				const submissionPromises = [];
+				studentIds.forEach((studentId) => {
+					submissionPromises.push(
+						testObjects.createTestSubmission({
+							homeworkId,
+							studentId,
+							comment: 'I dont know the answer',
+							submitted: true,
+							graded: true,
+						})
+					);
+				});
+				await Promise.all(submissionPromises);
+			};
+			it('does not paginate the submissions', async () => {
+				const { _id: teacherId } = await testObjects.createTestUser({ roles: ['teacher'] });
+				const student = await testObjects.createTestUser({ roles: ['student'] });
+				const otherStudentPromises = [];
+				for (let i = 0; i < 25; i += 1) {
+					otherStudentPromises.push(testObjects.createTestUser({ roles: ['student'] }));
+				}
+				let otherStudents = await Promise.all(otherStudentPromises);
+				otherStudents = otherStudents.map((s) => s._id);
+				const { _id: courseId } = await testObjects.createTestCourse({
+					teacherIds: [teacherId],
+					userIds: [student._id, ...otherStudents],
+				});
+
+				const homeworkPromises = [];
+				for (let i = 0; i < 50; i += 1) {
```

Comment:
^ see above, for 50 here

Reply:
same here @EzzatOmar  or @virgilchiriac

Reply:
but this not part of our code, these are old. ("ignore whitespace "setting?)